cmd/root.go

```diff
@@ -411,14 +413,20 @@ func expandEnvCmdObjs(cmd *cobra.Command, env string, components []string, cwd m
 	if err != nil {
 		return nil, err
 	}
-	params := importParams(string(envParamsPath))
-	expander.ExtCodes = append([]string{baseObj, params}, expander.ExtCodes...)
+
+	//
+	// Set up ExtCodes to resolve runtime variables such as the environment namespace.
+	//
+
+	params := importParams(string(paramsPath))
+	spec := importEnv(string(specPath))
```

Comment:
Ah, so, based on the comments in the design proposal, I assumed we were _not_ exposing the server URI or the env name, and instead exposing just the namespace for now?

Reply:
Ah sorry, I think this is the one point that I forgot to address in the proposal. The env `spec.json` doesn't contain the name so we would only be exposing the server URI and the namespace.

If I recall correctly, your reservations around exposing the server URI was the `if server == x else` problem. This is the same anti-pattern with namespace and I think it can be discouraged (as mentioned in our other discussion). I also think this pattern will be a lot less seen if `name` isn't exposed. 

Agree `name` is also a ksonnet CLI specific attribute so it makes little sense to expose at the prototype level but `server` and `namespace` are Kubernetes attributes that can be useful as args to ksonnet-lib APIs.

Reply:
Ok, that makes sense to me. I'm going to mark as "approve", let's see what @bryanl thinks.